File: backend/lib/L1_domain/entities/goals/goal_report.py
#  Copyright (c) 2022. Alexandr Moroz

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass
class GoalReport:
    eta_date: datetime | None = None
    fact_speed: float | None = 0
    plan_speed: float | None = 0

File: frontend/lib/L1_domain/entities/goals/goal_report.dart
// Copyright (c) 2022. Alexandr Moroz

// TODO: сомнительный класс. Возможно, он вовсе не понадобится. Сделано было для распространения логики расчёта скоростей... На разные клиенты
//  С другой стороны, возможно, копипаст клиентской логики будет не таким уж и плохим решением. Особенно с учётом экономии ресурсов сервера для расчётов.

class GoalReport {
  GoalReport({
    this.factSpeed = 0,
    this.planSpeed = 0,
    this.etaDate,
  });

  final num factSpeed;
  final num planSpeed;
  final DateTime? etaDate;
}
